src/panels/config/integrations/ha-integration-card.ts

```diff
@@ -310,15 +319,31 @@ export class HaIntegrationCard extends LitElement {
   }
 
   private _handleAction(ev: CustomEvent<ActionDetail>) {
-    const configEntry = ((ev.target as HTMLElement).closest("ha-card") as any)
-      .configEntry;
-    switch (ev.detail.index) {
-      case 0:
-        this._showSystemOptions(configEntry);
-        break;
-      case 1:
-        this._removeIntegration(configEntry);
-        break;
+    const thisCard = (ev.target as HTMLElement).closest("ha-card") as any;
+    const configEntry = thisCard.configEntry;
+    const canReload = thisCard.canReload;
+
+    if (canReload) {
```

Comment:
I didn't really like this solution, but I'm not familiar enough with `mwc` to come up with a better one.

Reply:
We should just figure out a better solution so that we don't have to rely on the index of the item. https://github.com/material-components/material-components-web-components/tree/master/packages/list#mwc-list-item-2

Reply:
Check <https://github.com/home-assistant/frontend/blob/dev/src/panels/lovelace/hui-root.ts#L491-L496>